openwisp_controller/api/serializers.py

```diff
@@ -0,0 +1,26 @@
+from swapper import load_model
+
+from openwisp_users.api.mixins import FilterSerializerByOrgOwned
+from openwisp_utils.api.serializers import ValidatedModelSerializer
+
+Template = load_model('config', 'Template')
+Vpn = load_model('config', 'Vpn')
+Device = load_model('config', 'Device')
+
+
+class TemplateSerializer(FilterSerializerByOrgOwned, ValidatedModelSerializer):
+    class Meta:
+        model = Template
+        fields = '__all__'
+
+
+class VpnSerializer(FilterSerializerByOrgOwned, ValidatedModelSerializer):
+    class Meta:
+        model = Vpn
+        fields = '__all__'
+
+
+class DeviceSerializer(ValidatedModelSerializer):
+    class Meta:
+        model = Device
+        fields = '__all__'
```

Comment:
all the templates will surely need to contain more logic

Reply:
Yes